weather_mv/loader_pipeline/sinks.py

```diff
@@ -72,15 +73,41 @@ def _make_grib_dataset_inmem(grib_ds: xr.Dataset) -> xr.Dataset:
     return data_ds
 
 
-def _preprocess_tif(ds: xr.Dataset, filename: str, tif_metadata_for_datetime: str) -> xr.Dataset:
+def match_datetime(file_name: str, regex_str: str) -> datetime.datetime:
+    """Extracts the datetime object from a string
+    """
+    char_to_replace = {
+        '%Y': '([0-9]{4})',
+        '%m': '([0-9]{2})',
+        '%d': '([0-9]{2})',
+        '%H': '([0-9]{2})',
+        '%M': '([0-9]{2})',
+        '%S': '([0-9]{2})',
+        '*': '.*'
+    }
+    for key, value in char_to_replace.items():
+        regex_str = regex_str.replace(key, value)
+    mtch = re.findall(regex_str, file_name)[0]
+    time_arr = list(map(int, mtch))
+    date_string = datetime.datetime(time_arr[0], time_arr[1], time_arr[2], time_arr[3], time_arr[4], time_arr[5])
+    return date_string
+
+
+def _preprocess_tif(ds: xr.Dataset, filename: str, tif_metadata_for_datetime: str, uri: str,
+                    band_names: t.Dict, initialization_time: str, forecast_time: str) -> xr.Dataset:
     """Transforms (y, x) coordinates into (lat, long) and adds bands data in data variables.
 
     This also retrieves datetime from tif's metadata and stores it into dataset.
     """
 
     def _get_band_data(i):
```

Comment:
Didn't we change this to a key-value pair instead of b0, b1, ...?

We are expecting `{band_name_in_ds: band_name_we_want}` from the json now.

Reply:
Resolved